src/Extension/KernelExtensions.cs

```diff
@@ -32,18 +34,62 @@ public static class KernelExtensions
 
         public static T UseProgressiveLearning<T>(this T kernel) where T : Kernel, IKernelCommandHandler<ParseNotebook>
         {
-            var argument = new Argument<FileInfo>("file");
-            
+            Option<Uri> fromUrlOption = new Option<Uri>(
+                "--from-url",
```

Comment:
is there any good way to test --from-url?

Reply:
What is the behavior you want to test?

Reply:
does #!start-lesson --from-url work?

Reply:
Are you looking to test the parsing behavior, the ability to retrieve the data from a URL, other details?

I think we have good examples in the tests for the `KeyValueStoreKernel`.